client/my-sites/sidebar/publish-menu.jsx

```diff
@@ -213,23 +223,30 @@ const PublishMenu = React.createClass( {
 
 		return (
 			<ul>
-				{ this.props.site && (
-					<QueryPostTypes siteId={ this.props.site.ID } />
+				{ this.props.siteId && (
+					<QueryPostTypes siteId={ this.props.siteId } />
 				) }
-				{ menuItems.map( this.renderMenuItem ) }
+				{ menuItems.map( this.renderMenuItem, this ) }
 			</ul>
 		);
 	}
-} );
+}
 
-export default connect( ( state ) => {
-	const siteId = get( getSelectedSite( state ), 'ID' );
+export default connect( ( state, { siteId } ) => {
 	const postTypes = getPostTypes( state, siteId );
 
 	return {
+		allSingleSites: areAllSitesSingleUser( state ),
+		canUser: ( cap ) => canCurrentUser( state, siteId, cap ),
```

Comment:
I'd like us to avoid instantiating functions in `mapState`. I'm wondering what the alternative could be here. Would it be conceivable to gather all the items, pluck their `capability`, `uniq` that collection, and return a dictionary for the current site? That is:

```js
const getCapabilities = ( state, siteId, menuItems ) =>
  transform( uniq( map( menuItems, 'capability' ) ), ( acc, cap ) => {
    acc[ cap ] = canCurrentUser( state, siteId, cap );
  } );

// then

return {
  allSingleSites: …,
  userCapabilities: getCapabilities( state, siteId ),
  isJetpack: …,
```

(It would be easier to cache this operation if `menuItems` wasn't rebuilt on every render from two different methods.)